### PR TITLE
Add public methods and hide protocol

### PR DESCRIPTION
**Pull request by leplatrem**
Just a try: what do you think ?

### CODE REVIEW

**Comment by ametaireau** [comment]
+1 for me.

**Comment by leplatrem** [comment]
We should problably reorder the methods to put endpoints on top

File: cliquet/resource.py [hunk 1]
```diff
@@ -123,7 +124,167 @@ def raise_invalid(self, location='body', **kwargs):
...
+            the total number of records in the result set, and the next page
+            url.
+        :rtype: tuple
+        """
+        filters = self._extract_filters()
+        sorting = self._extract_sorting()
+        pagination_rules, limit = self._extract_pagination_rules_from_token(
+            sorting)
+
+        include_deleted = self.modified_field in [f[0] for f in filters]
```
Comment (leplatrem) [inline cliquet/resource.py]: I'm not totally convinced of having filters extraction here. If we mean to keep protocol stuff out of customization, this should probably go back to the endpoint code.

```python
def get_records(self, filters, sorting, pagination, limit, include_deleted):
       # Customize : disable sorting on certain fields
       sorting.remove(('age', 1))
       return super(..., ...).get_records(...)
```

**Comment by leplatrem** [comment]
Well actually @Natim you can merge this, since it doesn't target master. We'll continue the discussion in #29

**Pull request merged**